sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -216,57 +245,77 @@ def _from_generated(cls, definition):
 
 class _Operation(object):
     def __init__(self, **kwargs):
-        self.status = kwargs.get("status", None)
-        self.status_details = kwargs.get("status_details", None)
-        self.error = kwargs.get("error", None)
-        self.start_time = kwargs.get("start_time", None)
-        self.end_time = kwargs.get("end_time", None)
-        self.job_id = kwargs.get("job_id", None)
+        self._status = kwargs.get("status", None)
+        self._status_details = kwargs.get("status_details", None)
+        self._start_time = kwargs.get("start_time", None)
+        self._end_time = kwargs.get("end_time", None)
+        self._job_id = kwargs.get("job_id", None)
+
+        error = kwargs.get("error", None)
+        # responses can return an empty error object when there's no error
+        if error is None or error.code is None:
+            self._error = None
+        else:
+            self._error = ODataV4Format(vars(error))
```

Comment:
Small but important detail here: ODataV4Format expects the inner error to be labeled as `innererror`, but the error we get back labels it as `inner_error` and doesn't get applied to the resulting ODataV4Format object. Does it make sense to edit the label to be `innererror`, or would that be forcing a square peg into a round hole?

Reply:
ODataV4Format expects JSON deserialized to a dict but the Error class generated by autorest has already deserialized the JSON and Pythonified "innererror" to "inner_error". What do you get from `error.as_dict()`? Also, `ODataV4Format.innererror` should be a JSON dict as well--it's JSON dicts all the way down.

Reply:
`error.as_dict()` gives the same dictionary as `vars(dict)` -- we're still left with an `inner_error` key 😕

Reply:
Can you use the [`key_transformer` argument](https://github.com/Azure/msrest-for-python/blob/master/msrest/serialization.py#L278) to rename it?

Reply:
Okay nice, that does work with something like:
```python
def my_key_transformer(key, attr_desc, value):
    if key == "inner_error":
        return ("innererror", value)
    return (key, value)
```
for errors with no inner error (`error.inner_error = None`) and errors with JSON dict inner errors (they get recursively renamed)

Reply:
It looks like something like this would work as well:
```python
error.as_dict(key_transformer=lambda k, _, v: ("innererror", v) if k == "inner_error" else (k, v))
```